ktor-server/jvm/src/io/ktor/server/application/plugins/api/PluginPhases.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.application.plugins.api
+
+import io.ktor.server.application.*
+
+/**
+ * Callable object that provides handlers to extend handling of an application call.
+ **/
+public interface OnCall {
+    /**
+     * Define how processing an HTTP call should be modified by the current [ApplicationPlugin].
+     **/
+    public operator fun invoke(block: suspend CallContext.(ApplicationCall) -> Unit): Unit
+}
+
+/**
+ * Callable object that provides handlers to extend data receiving when handling an application call.
+ **/
+public interface OnCallReceive {
```

Comment:
Can be fun interface

Reply:
Yes, but:
a) In future we may add more (sub)phases
b) It is more consistent with `OnCallRespond` atm